pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java

```diff
@@ -0,0 +1,130 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.dataservice.client;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.primitives.Ints;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.pentaho.di.core.database.DatabaseInterface;
+import org.pentaho.di.core.database.DatabaseMeta;
+import org.pentaho.di.core.plugins.DatabaseMetaPlugin;
+import org.pentaho.di.core.row.ValueMetaInterface;
+import org.pentaho.di.trans.dataservice.jdbc.ThinDriver;
+
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.emptyArray;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * @author nhudak
+ */
+@RunWith( MockitoJUnitRunner.class )
+public class DataServiceClientPluginTest {
+
+  DataServiceClientPlugin clientPlugin;
+
+  @Before
+  public void setUp() throws Exception {
+    clientPlugin = new DataServiceClientPlugin();
+
+    DatabaseMetaPlugin annotation = DataServiceClientPlugin.class.getAnnotation( DatabaseMetaPlugin.class );
+    clientPlugin.setPluginId( annotation.type() );
+    clientPlugin.setPluginName( annotation.typeDescription() );
+  }
+
+  @Test
+  public void testSupportChecks() throws Exception {
+    ImmutableSet<String> supported = ImmutableSet.of(
+      "supportsBooleanDataType", "supportsOptionsInURL", "supportsErrorHandling", "supportsSetMaxRows",
+      "supportsSetLong", "supportsTimeStampToDateConversion"
+    );
+
+    for ( Method method : DatabaseInterface.class.getMethods() ) {
+      String methodName = method.getName();
+      if ( methodName.startsWith( "supports" ) && method.getReturnType() == Boolean.TYPE ) {
+        assertThat( methodName, (Boolean) method.invoke( clientPlugin ), is( supported.contains( methodName ) ) );
+      }
+    }
+
+    assertThat( clientPlugin.isFetchSizeSupported(), is( false ) );
+  }
+
+  @Test
+  public void testDDL() throws Exception {
+    assertThat( clientPlugin.getCreateTableStatement(), is( "// Unsupported" ) );
+    assertThat( clientPlugin.getTruncateTableStatement( "dataService" ), is( "// Unsupported" ) );
```

Comment:
I've added my comment at the end of the list, but there are some in it that are not tested.